src/app/model/MoveResult.ts

```diff
@@ -1,7 +1,8 @@
+import { PawnWasPromoted } from './PawnWasPromoted';
 import { PieceWasMoved } from './PieceWasMoved';
 import { PieceWasCaptured } from './PieceWasCaptured';
 import { KingWasChecked } from './KingWasChecked';
 import { KingWasUnchecked } from './KingWasUnchecked';
 import { PawnPromotionWasEnabled } from './PawnPromotionWasEnabled';
 
-export type MoveResult = PieceWasMoved | PieceWasCaptured | KingWasChecked | KingWasUnchecked | PawnPromotionWasEnabled;
+export type MoveResult = PieceWasMoved | PieceWasCaptured | KingWasChecked | KingWasUnchecked | PawnPromotionWasEnabled | PawnWasPromoted;
```

Comment:
To nie jest MoveResult - PawnWasPromoted 
To jest osobna operacja. Bo move nie zwroci nigdy PawnWasPromoted
Mozna zrobic:
`export type ChessEvent = MoveResult | PawnWasPromoted`

Reply:
zrobiłem trochę inaczej :-)